docs/proposals/non-default-namespace-deployment.md

```diff
@@ -0,0 +1,145 @@
+   necessary permissions to the corresponding csi's service accounts.
+
+## Compatibility
+
+For the second approach we are binding directly to helm lifecycle management mechanisms, and there may be the problem 
+in case of possible supporting other configuration managers such as kustomize if there will be such requirement.
+
+## Implementation
+
+#### Necessary permissions for scc granting service account (operator's one or helm's post-install hook)
+1. In order to give permissions to certain service accounts, the granting user/service account should have been bound 
+   with the following role and rolebinding:
+```yaml
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+   name: role
+   namespace: test-csi
+rules:
+   - apiGroups:
+        - "rbac.authorization.k8s.io"
+     resources:
+        - rolebindings
+        - roles
+     verbs:
+        - get
+        - create
+        - update
+   - apiGroups:
+        - "security.openshift.io"
+     resourceNames:
+        - privileged
+     resources:
+        - securitycontextconstraints
+     verbs:
+        - use
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+   name: rolebinding
+   namespace: test-csi
+roleRef:
+   apiGroup: rbac.authorization.k8s.io
+   kind: Role
+   name: role
+subjects:
+   - kind: ServiceAccount
+     name: assignment
+     namespace: test-csi
+```
+2. The corresponding role should be persisted at helm charts and be bound dynamically to csi-node-sa and 
+   csi-baremetal-extender-sa service accounts right after their creation:
+```yaml
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+   name: role-scc
+   namespace: test-csi
+rules:
+   - apiGroups:
+        - "security.openshift.io"
+     resourceNames:
+        - privileged
+     resources:
+        - securitycontextconstraints
+     verbs:
+        - use
+```
+
+#### Moving creation csi's sa logic to csi-baremetal-operator
+1. Currently, service accounts are persisting at helm charts. In the considering approach csi's service accounts should be
+   created:
+   1. csi-node-sa: right before node daemonset creation (https://github.com/dell/csi-baremetal-operator/blob/master/pkg/node/node.go#L38);
+   2. csi-baremetal-extender-sa: right before scheduler extender creation (https://github.com/dell/csi-baremetal-operator/blob/master/pkg/scheduler_extender.go#L38);
+   3. other service accounts should be created by the analogue.
+2. Right after csi-node-sa/csi-baremetal-extender-sa creation, operator should create additional rolebindings as described
+   in the previous section and also create the current needed ones.
+3. After the above preparations daemonsets can be created like before. 
+
+#### Implementing separate helm post-install hook
+1. For separate post-install hook creation we should implement the separate component (e.g. _csi-postconfigurator_), which will be run as a separate job
```

Comment:
Will CSI pods be restarting due to lack of access until this job completed?

Reply:
The answer is at 136 line:
https://github.com/dell/csi-baremetal/pull/645/files#diff-6d7d994f70a983b2f669ca3449d6d261007f2483daf9312a0190a8f36bf55ce0R136